wagtail/contrib/modeladmin/views.py

```diff
@@ -254,7 +256,8 @@ class IndexView(SpreadsheetExportMixin, WMABaseView):
     SEARCH_VAR = 'q'
     ERROR_FLAG = 'e'
     EXPORT_VAR = 'export'
-    IGNORED_PARAMS = (ORDER_VAR, ORDER_TYPE_VAR, SEARCH_VAR, EXPORT_VAR)
+    LANG_VAR = 'lang'
```

Comment:
you need to filter the queryset based on `LANG_VAR` (after https://github.com/wagtail/wagtail/blob/672482ee09663e62df138901c375d9988d72276e/wagtail/contrib/modeladmin/views.py#L297) to actually filter things.

I would also change from `lang` to `locale` to be consistent with the other places (e.g. snippets)

Reply:
Hmmm, is that the use of that `LANG_VAR`? I just copied it without understanding it. hehe

Now that you said that, I don't think we need the select or to filter the queryset here, because this PR already adds this (copied from #6230):
```        
if issubclass(self.model, TranslatableMixin):
    list_filter += ('locale', )
```
Which gives us this:

![image](https://user-images.githubusercontent.com/937689/149414384-d76c49f8-b1f4-4e21-83fb-c79a78beedfd.png)

Unlike the snippets, the modeladmin already has this type of filter, so it seems better to use it.

So what is still missing now is a way to set the locale when adding a new item. Without it, it's always using the same locale, without asking. But I'm not sure how you would like it implemented. Maybe a select in the top bar like in the snippets?

Reply:
yes, create/edit should follow the pattern used by snippets (selector in the top bar, handle the `locale` get param)